### ISSUE TITLE
[clang-tidy] readability-magic-numbers cannot handle `long double`

### ISSUE THREAD

**Issue by carlosgalvezp** [llvm/llvm-project#61656]
Example:

```cpp
void foo(long double);

int main()
{
    foo(1.0L);
}
```

Leads to:

```console
$ clang-tidy -checks=-*,readability-magic-numbers main.cpp

warning: 1.0L is a magic number; consider replacing it with a named constant [readability-magic-numbers]
foo(1.0L);
        ^
```
[Repro on Compiler Explorer](https://godbolt.org/z/cMEfefabY).

This is problematic when one has user-defined floating literals, which as required by the standard must have a `long double` as input:

```cpp
float operator ""_meters(long double x)
{
    return static_cast<float>(x);
}

void foo(float);

int main()
{
  foo(1.0_meters);  // Warning here
}
```

**Comment by ustchcs-bugfinder** [comment]
Actually user-defined literals using "literal operators" don't have a meaningful name. 

In this case, "1.0_meters" do show us the length unit "meter". But 1.0 is still a magic number.

**Comment by carlosgalvezp** [comment]
Yes it is, but should not be warned about, since by default 1.0 is whitelisted. If I used a regular 1.0 (float or double) clang-tidy wouldn't complain.

**Comment by PiotrZSL** [comment]
Not so easy to implement, because half of llvm classes does not support long double.
